### PR TITLE
added workflows and git hooks

### PR DESCRIPTION
**Pull request by Ellpeck**

### CODE REVIEW

**Comment by EagleoutIce** [comment]
We should definitely outsource this in the future, and probably even start adding references/clear links to parts in the flower documentation (/whatever part deploys then)

**Comment by Ellpeck** [comment]
I forgot to npm install, i think i might die rn

**Comment by Ellpeck** [comment]
> We should definitely outsource this in the future, and probably even start adding references/clear links to parts in the flower documentation (/whatever part deploys then)

Okay, I'll reference this in the [flowR issue](https://github.com/Code-Inspect/flowr/issues/699) as well.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        uses: actions/checkout@v4
+      - name: Install node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 21.6.x
+          registry-url: 'https://registry.npmjs.org/'
+      - name: Install dependencies
+        run: npm ci
+      - name: Test
+        uses: coactions/setup-xvfb@v1
```
Comment (EagleoutIce) [inline .github/workflows/test.yml:21]: Pls add a comment stating the reason for this

Comment (Ellpeck) [reply .github/workflows/test.yml:21]: done

**Pull request merged**